### PR TITLE
Feature: Add INPUT parameter `bands_to_print` for LCAO basis set

### LINKED ISSUE #2924: nbands_istate: This parameter may be changed to select the i-th band to calculate.
### Background

In some cases, people want to know the charge density distribution of deeper energy levels,such as the twisted systems.

However, these systems are usually large, and the calculation of the partial charge density of multiple bands will greatly consume storage space.

### Describe the solution you'd like

It is suggested to add a setting: one can directly select the specific band index for get_pcha or get_wf.

### Additional Context

_No response_

### Task list for Issue attackers

- [x] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team and stakeholders on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

### LINKED ISSUE #3947: Band-decomposed (partial) charge densities using PW basis set appear unnormalized
### Describe the bug

While performing band-decomposed (partial) charge density calculations using the PW basis set, I've observed that the charge density integrals are not normalized and are incrementally increasing across bands (using a simple script written by Python attached below). This trend suggests that there might be an issue with array reinitialization in the subsequent computational steps:
> $ python3 grid_integral.py
电荷密度积分：band1.cube 1.9999983547907862
电荷密度积分：band2.cube 3.999996709581772
电荷密度积分：band3.cube 5.999995064372504
电荷密度积分：band4.cube 7.999993419163174
电荷密度积分：band5.cube 9.999991773953775
电荷密度积分：band6.cube 11.999990128744749
电荷密度积分：band7.cube 13.999988483535311
电荷密度积分：band8.cube 15.999986838326754

### Expected behavior

- For LCAO (one might have to add additional INPUT parameters such as `calculation=get_pchg`), the integral values should remain consistent and close to `2`. Here's one of my results:
> 电荷密度积分：BAND1_SPIN0_CHG.cube 1.9999983010347646
电荷密度积分：BAND2_SPIN0_CHG.cube 1.9999983636754213
电荷密度积分：BAND3_SPIN0_CHG.cube 1.9999982713382998
电荷密度积分：BAND4_SPIN0_CHG.cube 1.9999982579638917

- For PW, note the increasing trend in the integral values across bands, which is unexpected and indicates a possible bug.

#### Environment Setup
- **Input Files**: Provided in the attachments, including `INPUT`, `STRU`, and a Python script `grid_integral.py`.
- **Script Functionality**: `grid_integral.py` is designed to compute the total space integral of charge density from `.cube` files. A result close to `2` is expected when `nspin=1`.

#### Steps to Reproduce
1. Run ABACUS using the provided `INPUT` and `STRU` files.
2. Execute the `grid_integral.py` script to calculate the charge density integrals.

[Input files.zip](https://github.com/deepmodeling/abacus-develop/files/14929538/Input.files.zip)

### Environment

_No response_

### Additional Context

I suspect that this bug was introduced in PR #3501. I plan to fix it along with PR #3934.

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

### PR DESCRIPTION
**Pull request by AsTonyshment**
### Linked Issue
Fixes #2924 and #3266. Updates: Fixes #3947.
Partially implements feature support #3572.

### What's changed?
- Introduced a new INPUT parameter `bands_to_print` for **LCAO** basis set, akin to `ocp_set`. This allows for selective band calculation when the `calculation` keyword is set to `get_pchg`.
- Retained `nbands_istate` temporarily as `calculation = get_wf` remains unchanged.
- Updated and expanded the related documentation.
- Resolved the band index discrepancy in the PW basis set, ensuring that indexing now consistently starts from 1 for users.

### Notice
May potentially conflict with the naming in PR #3501 (**PW** basis set), which will be addressed and unified in subsequent updates.

Updates: Following @mohanchen's suggestion, I have standardized the INPUT parameter to `bands_to_print`, and uniformly changed the input format to string. This has been updated accordingly in the PW basis set code. The docs has also been updated to reflect these changes. INPUT parameter `band_print_num` is deleted.

### CODE REVIEW

File: docs/advanced/input_files/input-main.md [hunk 1]
```diff
@@ -1680,6 +1674,20 @@ These variables are used to control the output of properties.
 - **Description**: Generate output files used in rpa calculations.
 - **Default**: False
 
+### nbands_istate
+
+- **Type**: Integer
+- **Availability**: Only used when `calculation = get_wf` or `calculation = get_pchg`.
+- **Description**: The number of bands around the Fermi level you would like to calculate. `get_wf` means to calculate the envelope functions of wave functions $\Psi_{i}=\Sigma_{\mu}C_{i\mu}\Phi_{\mu}$, where $\Psi_{i}$ is the ith wave function with the band index $i$ and $\Phi_{\mu}$ is the localized atomic orbital set. `get_pchg` means to calculate the density of each wave function $|\Psi_{i}|^{2}$. Specifically, suppose we have highest occupied bands at 100th wave functions. And if you set this variable to 5, it will print five wave functions from 96th to 105th. But before all this can be carried out, the wave functions coefficients  should be first calculated and written into a file by setting the flag `out_wfc_lcao = 1`.
```
Comment (mohanchen) [inline docs/advanced/input_files/input-main.md]: an extra space here needs to be removed

File: source/module_esolver/esolver_ks_lcao_elec.cpp [hunk 2]
```diff
@@ -408,6 +408,8 @@ void ESolver_KS_LCAO<TK, TR>::others(const int istep)
                   GlobalV::NBANDS,
                   GlobalV::nelec,
                   GlobalV::NSPIN,
+                  GlobalV::NLOCAL,
+                  GlobalV::OUT_BAND_INDEX,
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao_elec.cpp]: do not add new GlobalV parameter

**Comment by WHUweiqingzhou** [comment]
Is it possible to use same input parameters for PW and LCAO?

**Comment by AsTonyshment** [comment]
> Is it possible to use same input parameters for PW and LCAO?

Yes, this is what I'm doing at present.

**Review by mohanchen — APPROVED** [review]
LGTM

**Pull request merged**